### DISCUSSION TITLE
Installing util.shape

### DISCUSSION THREAD

**Discussion by DprOliv** [IfcOpenShell/IfcOpenShell#4930] [General]
Does the usage of ifcopenshell.util.shape requires installing PythonOCC ? 
I already have ifcopenshell.util .... properly installed   (used Conda) .... 
Thanks for any clarificaiton .....

**Comment by Moult** [comment]
No, it doesn't require PythonOCC. Some functions require shapely, though.

**Reply by DprOliv** [reply]
Thanks @Moult 
But i think i still need to install something else... right ? 
Because, when i try to import ifcopenshell.util.shape .. i've got an error "ModuleNotFoundError: No module named 'ifcopenshell.util.shape'

But i can import ifcopenshell.util normally ....

I have a conda environment  and have ifcopenshell installed in it ... 
Python 3.10.6 running in a MacOS 12.6.6 (Monterey) ...
I do have Shapely installed too

**Reply by Moult** [reply]
That sounds like you have an outdated version of IfcOpenShell. Can you try upgrade? Does ifcopenshell.util.shape exist at all in your site packages?

**Reply by DprOliv** [reply]
Yes @Moult ...... That's the case! In my site packages there's no shape.py inside utils folder...
I run Conda install trying to update my ifcOpenShell .....

Can i just copy the file shape.py (any other ?) from GitHub  ( into my site pacakge/ifcopenshell ...)  
(https://github.com/IfcOpenShell/IfcOpenShell/blob/v0.8.0/src/ifcopenshell-python/ifcopenshell/util/shape.py)  
into my  site-package/util folder ?

Thank you

**Reply by DprOliv** [reply]
I tried to copy shape.py but it shows errors when imported.
My ifcOpenShell version is 0.7..0 (v0.7.0-1b1fd1e6)
Can i just get the files in GitHub and download all files ... would that work?
If not... If i use installation via Conda would the actual 0.8.0 be installed ?

Thanks

**Reply by Moult** [reply]
Feel free to update via conda. There are a number of known bugs as we transition to v0.8.0 but if you're OK with that it would help us with testing.

If not OK with that, I'm pretty hopeful that there is a version in between 0.7.0 and 0.7.9 that has util.snape but isn't yet using the buggy ifcopenshell core on conda. Perhaps @Krande knows?

**Reply by DprOliv** [reply]
@Moult 
Thanks for your recommendations and advice.

instead of using Conda Install...  i could solve the issue here by downliading the zip package... 
Here's what i did: 
- I deleted the old ifcOpenShell from site-packages folder... 
- I downloaded the zip package for Python 3.10 for MacOS Intel 64 bits 
(https://github.com/IfcOpenShell/IfcOpenShell/releases/download/ifcopenshell-python-0.7.10/ifcopenshell-python-0.7.10-py310-macos64.zip).
- In order to avoid the OS X blocking the package (due to the OS X quarantine ... which prevents the the wrapper "_ifcopenshell_wrapper.cpython-310-darwin.so" to run when imported in Python i run: "xattr -d com.apple.quarantine <<zipfile.zip>>"
- Finally i copied the ifcopenshell unziped and clear from quarantine to site-packages folder in my Python folder...

util.shape can be imported now.
